### ISSUE TITLE
Dynamic channels should have SHOW_PROTOCOL set

### ISSUE THREAD

**Issue by hardening** [ogon-project/ogon-project#11]
If we wanna be microsoft compatible, dynamic channels should have the SHOW_PROTOCOL set, and have  CHANNEL_PDU_HEADER before the data (https://docs.microsoft.com/en-us/windows/win32/api/wtsapi32/nf-wtsapi32-wtsvirtualchannelread).
Changing this may have impacts on ogon channels implementations.

**Comment by hardening** [comment]
https://github.com/ogon-project/ogon/pull/22 implements this behaviour.

**Comment by nfedera** [comment]
The `CHANNEL_FLAG_SHOW_PROTOCOL` is only used in static channels.
For dynamic channels there is no way for the plugin (rdp client side) to control if the server side `WTSVirtualChannelRead` implementation should skip or include the `CHANNEL_PDU_HEADER`.
The specs only says that "_The application endpoint MUST be given enough information to perform reassembly_".

So this is really implementation specific and yes, the Microsoft implementations always include this struct for dynamic channels, as you've pointed out.

For ogon this data would have to be injected in the `wts_read_drdynvc_data_first` and `wts_read_drdynvc_data` functions of [rdp-server/channels.c](https://github.com/ogon-project/ogon/blob/master/rdp-server/channels.c)

**Comment by hardening** [comment]
@nfedera it looks like the doc says explicitly that the header is present ẁith dynamic channels => https://docs.microsoft.com/en-us/windows/win32/api/wtsapi32/nf-wtsapi32-wtsvirtualchannelread (`Buffer` parameter)

**Comment by nfedera** [comment]
Right, the API docs ... I had the protocol spec in mind.
FreeRDP's implementation of [WTSVirtualChannelRead](https://github.com/FreeRDP/FreeRDP/blob/8665ac7e81ac9f00f945de3cb8f0be877b0db677/libfreerdp/core/server.c#L1220)  also seems to lack this behaviour.

**Comment by hardening** [comment]
@nfedera well the win32 implementation is correct regarding this behaviour, but not the ogon WTSAPI lib... Doing the change would mean adjusting all server-side dynamic channel implementations (at least rdpei for ogon scope)

**Comment by nfedera** [comment]
@hardening true, and since rdpei in xserver-ogon uses the FreeRDP server channel implementation (which does also not expect these headers) we cannot do that change until this is changed/fixed in FreeRDP.
